float32 y_coord
float32 z_coord
float32 dis

---